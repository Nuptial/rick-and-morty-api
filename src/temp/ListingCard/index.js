import {
  Grid,
  Typography,
  Card,
  CardContent,
  CardMedia,
  CardActionArea,
} from "@mui/material";
import { useNavigate } from "react-router-dom";

const ListingCard = ({ character }) => {
  const navigate = useNavigate();

  if(!character) {
    return false;
  }

  const routeChange = () => {
    let path = `/character/${character.id}`;
    navigate(path);
  };

  return (
    <Grid item xs={2} sm={4} md={2} key={character.id}>
      <Card style={{ cursor: "pointer", boxShadow: "0 8px 40px -12px rgba(0,0,0,0.3)" }} onClick={routeChange}>
        <CardActionArea>
          <CardMedia
            component="img"
            height="250"
            image={character.image}
            alt={character.name}
          ></CardMedia>
          <CardContent>
            <Typography align="center" className="character-name">
              {character.name}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </Grid>
  );
};

export default ListingCard;
