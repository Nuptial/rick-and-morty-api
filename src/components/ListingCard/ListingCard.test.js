import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import ListingCard from '../ListingCard';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<ListingCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a single item', () => {
    const character = {
        "__typename": "Character",
        "id": "18",
        "name": "Antenna Morty",
        "status": "Alive",
        "species": "Human",
        "type": "Human with antennae",
        "gender": "Male",
        "image": "https://rickandmortyapi.com/api/character/avatar/18.jpeg",
        "created": "2017-11-04T22:25:29.008Z",
        "origin": {
            "__typename": "Location",
            "id": null,
            "name": "unknown"
        },
        "location": {
            "__typename": "Location",
            "id": "3",
            "name": "Citadel of Ricks"
        },
        "episode": [
            {
                "__typename": "Episode",
                "id": "10",
                "name": "Close Rick-counters of the Rick Kind"
            },
            {
                "__typename": "Episode",
                "id": "28",
                "name": "The Ricklantis Mixup"
            }
        ]
    }
    const tree = renderer.create(<BrowserRouter><ListingCard character={character} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});