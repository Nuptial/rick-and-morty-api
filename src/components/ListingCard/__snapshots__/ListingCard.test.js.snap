// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `null`;

exports[`renders correctly when there is a single item 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-2 css-13i4rnv-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    onClick={[Function]}
    style={
      Object {
        "boxShadow": "0 8px 40px -12px rgba(0,0,0,0.3)",
        "cursor": "pointer",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="Antenna Morty"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="250"
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter character-name css-1uwgr7b-MuiTypography-root"
        >
          Antenna Morty
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
